woocommerce/woocommerce
# Fixed Asset Output Caching

### All Submissions:






### Changes proposed in this Pull Request:

Something that was highlighted by #34502 is that the build outputs for `woocommerce/client/admin` and `woocommerce/client/legacy` are not being cached as expected. After the build has been "cached", if you delete the assets from `plugins/woocommerce/assets`, they won't be restored by a subsequent `build`. It will see these as having been cached, not rebuild them, but _also_ not restore the cached files.

This problem manifests in CI in particular because in the above PR it is relying on the cached build outputs being restored in order to function. Since those assets aren't cached in the first place, the plugin is broken and does not function.

This pull request:

* Updates Turborepo to the latest version to grab bug fixes and changes around globbing.
* Changed the globs in `turbo.json` for compatibility with the latest Turborepo.
* Adjusts the `turbo.json` pipelines so that the build outputs are cached correctly.

### How to test the changes in this Pull Request:

1. Clear out `node_modules/.cache` entirely so that you're working fresh.
2. Run `rm -rf plugins/woocommerce/assets/js plugins/woocommerce/assets/css plugins/woocommerce/assets/client/admin ` to remove the built assets
3. Run `pnpm run build` and let all of the assets build. Run it again after and confirm that we're utilizing the cached assets.
4. Run `rm -rf plugins/woocommerce/assets/js plugins/woocommerce/assets/css plugins/woocommerce/assets/client/admin ` to remove the built assets again.
5. Run `pnpm run build` again. It should utilize the cache in the same way that the third step did. Confirm that it restores the files in `plugins/woocommerce/assets` correctly. Make sure WooCommerce works as expected!

### Other information:




### FOR PR REVIEWER ONLY:
